openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+        identically matching. This is a helper method for the can_merge(...)
+        and merge(...) methods.
+
+        Parameters
+        ----------
+        other : openmc.Tally
+            Tally to check for mergeable nuclides
+
+        """
+
+        no_nuclides_match = True
+        all_nuclides_match = True
+
+        # Search for each of this tally's nuclides in the other tally
+        for nuclide in self.nuclides:
+            if nuclide not in other.nuclides:
+                all_nuclides_match = False
+            else:
+                no_nuclides_match = False
+
+        # Search for each of the other tally's nuclides in this tally
+        for nuclide in other.nuclides:
+            if nuclide not in self.nuclides:
+                all_nuclides_match = False
+            else:
+                no_nuclides_match = False
+
+        # Either all nuclides should match, or none should
+        return no_nuclides_match or all_nuclides_match
+
+    def _can_merge_scores(self, other):
+        """Determine if another tally's scores can be merged with this one's
+
+        The scores between the two tallies must be mutually exclusive or
+        identically matching. This is a helper method for the can_merge(...)
+        and merge(...) methods.
+
+        Parameters
+        ----------
+        other : openmc.Tally
+            Tally to check for mergeable scores
+
+        """
+
+        no_scores_match = True
+        all_scores_match = True
+
+        # Search for each of this tally's scores in the other tally
+        for score in self.scores:
+            if score in other.scores:
+                no_scores_match = False
+
+        # Search for each of the other tally's scores in this tally
+        for score in other.scores:
+            if score not in self.scores:
+                all_scores_match = False
+            else:
+                no_scores_match = False
+
+            if score == 'current' and score not in self.scores:
+                return False
+
+        # Nuclides cannot be specified on 'flux' scores
+        if 'flux' in self.scores or 'flux' in other.scores:
+            if self.nuclides != other.nuclides:
+                return False
+
+        # Either all scores should match, or none should
+        return no_scores_match or all_scores_match
+
+    def can_merge(self, other):
+        """Determine if another tally can be merged with this one
+
+        If results have been loaded from a statepoint, then tallies are only
+        mergeable along one and only one of filter bins, nuclides or scores.
+
+        Parameters
+        ----------
+        other : openmc.Tally
+            Tally to check for merging
+
+        """
+
+        if not isinstance(other, Tally):
+            return False
+
+        # Must have same estimator
+        if self.estimator != other.estimator:
+            return False
+
+        equal_filters = sorted(self.filters) == sorted(other.filters)
```

Comment:
Just wanted to note that `sorted(list1) == sorted(list2)` is `n + m + log(n)*n + log(m)*m` time complexity. Could be a bottle neck if any of these fields are large.